src/main/java/com/deusgmbh/xcusatio/api/APIService.java

```diff
@@ -25,84 +22,196 @@
  * 
  *
  */
-
-// TODO refactoring, simplify access to APIs
 public abstract class APIService {
-
     private static final Logger LOGGER = Logger.getLogger(APIService.class.getName());
 
-    private String baseUrlText;
-    private URL baseUrl;
-    private HttpURLConnection connection;
-    private Document responseAsDocument;
-    private NodeList allDocumentNodes;
-    private String responseAsText;
+    private URL requestUrl;
+    private String responseAsJsonString;
+
+    public void getJsonStringFromInputStream() throws IOException {
+        InputStream is = this.requestUrl.openStream();
+        this.responseAsJsonString = IOUtils.toString(is, "UTF-8");
+        is.close();
+    }
 
     public abstract <T extends Object> T get(UserSettings usersettings);
 
-    public void requestWebsite() throws IOException {
-        try {
-            this.baseUrl = new URL(this.baseUrlText);
-        } catch (MalformedURLException e) {
-            LOGGER.warning("Malformed URL, got " + this.baseUrlText + " as input string.");
+    public abstract void buildRequestUrl(UserSettings usersettings);
+
+    public abstract void transmitDataToWebsite();
+
+    /* just for tests */
+    public abstract void printResponse();
+
+    /* differs in terms of placement of desired info inside XML/JSON */
+    public abstract void extractDesiredInfoFromResponse() throws JSONException, ParseException;
+
+    /* differs in terms of data format returned by the call */
+    public abstract void getResponseFromWebsite();
+
+    /**
+     * 
+     * @param into
+     *            provides the list to move into in order to get the target list
+     *            of json objects out of it
+     * @param KEY
+     *            represents the key referring to this target list of json
+     *            objects
+     * @return the target list
+     * @throws JSONException
+     */
+    protected List<JSONObject> goInside(List<JSONObject> into, String KEY) throws JSONException {

```

Comment:
So wie die nächsten Funktionen aussehen, hast du dir hier ne kleine lib zusammengeschrieben, die einfach nur dabei hilft JSONs auszulesen. Aber dafür gibt es doch bestimmt schon andere Frameworks, die das bereits unterstützen und auch besser machen:
 - [How to parse JSON in Java](https://stackoverflow.com/questions/2591098/how-to-parse-json-in-java)
- [Oracles Java API for JSON Processing](http://www.oracle.com/technetwork/articles/java/json-1973242.html)